### PR TITLE
wallet: clarify replace fields in help output

### LINKED ISSUE #27781: Unclear documentation about TX replacements in `gettransaction`
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behaviour

The two fields about TXID replacement in `gettransaction` currently says this: 
```
 "replaced_by_txid" : "hex",              (string, optional) The txid if this tx was replaced.
 "replaces_txid" : "hex",                 (string, optional) The txid if the tx replaces one.
```

From looking at the [code](https://github.com/bitcoin/bitcoin/blob/71300489af362c3fed4736de6bffab4d758b6a84/src/wallet/feebumper.cpp#L356) it appears these fields are only set if the TXs in question are replaced by the wallet itself. This was not obvious to me.

### Expected behaviour

I'd expect the documentation to mention that these fields are only set for transactions that originate _from_ the wallet, and not for transactions received _into_ the wallet. 

### Steps to reproduce

1. Receive a fee-bumped transaction to a Bitcoin Core wallet. 
2. Observe that calling `gettransaction` on that transaction does _not_ include the `replaced_by_txid` or `replaces_txid` fields. 

### Relevant log output

_No response_

### How did you obtain Bitcoin Core

Pre-built binaries

### What version of Bitcoin Core are you using?

v24.0.1

### Operating system and version

MacOS Ventura 13.2.1 

### Machine specifications

_No response_

### PR DESCRIPTION
**Pull request by torkelrogstad**
Closes #27781.

### CODE REVIEW

**Comment by brunoerg** [comment]
I suppose that changing them in `TransactionDescriptionString` will also affect `listsinceblock` and `listtransactions`. So, I think would be appropriate to update the PR title/commit message?

**Review by brunoerg — COMMENTED** [review]
Concept ACK

**Comment by torkelrogstad** [comment]
Thanks for pointing that out @brunoerg. Updated the commit message and PR title.

**Comment by luke-jr** [comment]
otoh, it would be nice to populate these when we detect a replaced receive tx...

File: src/wallet/rpc/transactions.cpp [hunk 1]
```diff
@@ -415,8 +415,8 @@ static std::vector<RPCResult> TransactionDescriptionString()
            {
                {RPCResult::Type::STR_HEX, "txid", "The transaction id."},
            }},
-           {RPCResult::Type::STR_HEX, "replaced_by_txid", /*optional=*/true, "The txid if this tx was replaced."},
-           {RPCResult::Type::STR_HEX, "replaces_txid", /*optional=*/true, "The txid if the tx replaces one."},
+           {RPCResult::Type::STR_HEX, "replaced_by_txid", /*optional=*/true, "Only if 'category' is 'send'. The txid if this tx was replaced."},
+           {RPCResult::Type::STR_HEX, "replaces_txid", /*optional=*/true, "Only if 'category' is 'send'. The txid if the tx replaces one."},
```
Comment (pinheadmz) [inline src/wallet/rpc/transactions.cpp:419]:
```suggestion
           {RPCResult::Type::STR_HEX, "replaces_txid", /*optional=*/true, "Only if 'category' is 'send'. The txid if this tx replaces another."},
```

**Review by pinheadmz — COMMENTED** [review]
concept ACK

Included a nit for better consistency with the messages. I'll also point out, regarding the original issue, that `gettransaction` also returns a `walletconflicts` field which is present for both senders and receivers.

**Comment by brunoerg** [comment]
Are you still working on it?

**Pull request closed without merging**